### ISSUE TITLE
Demonstrate simple terminal user interface

### ISSUE THREAD

**Issue by whereswaldon** [LinuxAtApp/matterleast#3]
Create a subpackage `github.com/LinuxAtApp/matterleast/spikes/termui`
Using the [termbox](https://github.com/nsf/termbox-go) package or similar, create a full-screen terminal application with a scrolling text area and a constant text-entry area. They can be populated with random data, it's the UI itself that we care about.

**Comment by camwood23** [comment]
Added some simple bots to test multiple users adding messages. Needs to be tested with users sending messages from the back end though.

**Comment by whereswaldon** [comment]
@camwood23 What do you mean by that testing statement. I'm not sure that I understand.

**Comment by camwood23** [comment]
We need to make sure that this method of adding messages still works with what you guys are writing. I'm mostly just not sure how you guys plan to be receiving and storing messages.

**Comment by whereswaldon** [comment]
@camwood23 For the time being, assume you can call some methods like `GetMessage() string` and `SendMessage(message string)` that make that work. Maybe I'm misunderstanding the issue though. What exactly do you mean by "adding messages"? Is that about displaying them, or sending them to the server?

**Comment by camwood23** [comment]
Yeah, I was thinking that I could just call the SendMessage whenever the user tries to send a message. But I also need to keep track of what messages need to be displayed every time the screen updates. So right now I have a struct that keeps a log of every message sent to it by different users. And I wasn't sure if either the back end would have it's own struct where it manages what messages are where and then the front end would just access that struct whenever it updates the screen. Or if the front end would need to control the struct and the back end will just send it messages whenever a message is available.... I am mostly just wondering how my project is going to be communicating with every one else's project.

Edit: Although my code is in a functional state if you guys want to use it.. It is currently just keeping track of the log itself and all you have to do is call the ReceiveMessage function with a message and author. You can also add a call to handle_enter to send message ( I apologize for not informing everyone of that sooner )

**Comment by whereswaldon** [comment]
@camwood23 I think having a `RecieveMessage` function is a fine way to handle things. Is this code ready to be merged in as a spike? It doesn't need to integrate with anything else for that. It will just serve as a reference for how to write the UI

**Comment by camwood23** [comment]
Yeah, you can merge if you would like to!
